File: Plugins/Kingbotk/AWB Plugin/LED.vb
Imports System.Drawing

Namespace AutoWikiBrowser.Plugins.SDKSoftware.Kingbotk.Components
    Friend Enum Colour
        Red
        Green
        Blue
    End Enum
    ''' <summary>
    ''' A simple "LED" user control
    ''' </summary>
    ''' <remarks></remarks>
    Friend NotInheritable Class LED
        Private br As Brush = Brushes.Red, col As Colour = Colour.Red

        Protected Overrides Sub OnPaint(ByVal e As System.Windows.Forms.PaintEventArgs)
            MyBase.OnPaint(e)
            e.Graphics.FillEllipse(br, 2, 2, Me.Size.Width - 2, Me.Size.Height - 2)
        End Sub
        Public Property Colour() As Colour
            Get
                Return col
            End Get
            Set(ByVal value As Colour)
                Select Case col
                    Case Colour.Blue
                        br = Brushes.Blue
                        col = Colour.Blue
                    Case Kingbotk.Components.Colour.Green
                        br = Brushes.LimeGreen
                        col = Colour.Green
                    Case Kingbotk.Components.Colour.Red
                        br = Brushes.Red
                        col = Colour.Red
                End Select

                MyBase.Refresh()
            End Set
        End Property
    End Class
End Namespace

File: tags/REL_4_4_1/WikiFunctions/Controls/LED.cs
/*
(C) 2007 Stephen Kennedy (Kingboyk) http://www.sdk-software.com/

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation; either version 2 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program; if not, write to the Free Software
Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
*/

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Text;
using System.Windows.Forms;

namespace WikiFunctions.Controls
{
    public enum Colour { Red, Green, Blue };

    /// <summary>
    /// A simple "LED" control
    /// </summary>
    public partial class LED : UserControl
    {
        private Brush br = Brushes.Red;
        private Colour col = Colour.Red;

        public LED()
        {
            InitializeComponent();
        }

        protected override void OnPaint(System.Windows.Forms.PaintEventArgs e)
        {
            base.OnPaint(e);
            e.Graphics.FillEllipse(br, 2, 2, this.Size.Width - 2, this.Size.Height - 2);
        }

        public Colour Colour
        {
            get { return col; }
            set
            {
                switch (value)
                {
                    case Colour.Blue:
                        br = Brushes.Blue;
                        break;
                    case Colour.Green:
                        br = Brushes.Green;
                        break;
                    case Colour.Red:
                        br = Brushes.Red;
                        break;
                }
                col = value;

                base.Refresh();
            }
        }
    }
}
